src/components/Chat/ChatsSidebar.tsx

```diff
@@ -8,19 +8,43 @@ interface ChatListProps {
   currentChatHistory: ChatHistory | undefined;
   onSelect: (chatID: string) => void;
   newChat: () => void;
+  setChatHistoriesMetadata: (chat: ChatHistoryMetadata[]) => void;
+  setShowChatbot: (showChat: boolean) => void;
 }
 
 export const ChatsSidebar: React.FC<ChatListProps> = ({
   chatHistoriesMetadata,
   currentChatHistory,
   onSelect,
   newChat,
+  setChatHistoriesMetadata,
+  setShowChatbot,
 }) => {
   const [reversedChatHistoriesMetadata, setReversedChatHistoriesMetadata] =
     useState<ChatHistoryMetadata[]>([]);
   useEffect(() => {
     setReversedChatHistoriesMetadata(chatHistoriesMetadata.reverse());
   }, [chatHistoriesMetadata]);
+  const currentSelectedChatID = useRef<string | undefined>();
+
+  useEffect(() => {
+    const deleteChatRow = window.ipcRenderer.receive(
+      "remove-chat-at-id",
+      (chatID) => {
+        const filteredData = chatHistoriesMetadata.filter(item => item.id !== chatID)
+        setChatHistoriesMetadata(filteredData.reverse());
+        if (chatID === currentSelectedChatID.current) {
+          setShowChatbot(false);  
+        }
+        window.electronStore.updateAllChatHistories(chatID);
```

Comment:
Nit but we should probably update the store first and then update the renderer process' record of the chat metadata